client/src/App.jsx

```diff
@@ -2,18 +2,19 @@ import React, { useState } from 'react';
 
 import { BrowserRouter as Router, Route } from 'react-router-dom';
 
-import UserContext from './context/UserContext';
+import { UserContext } from './context/UserContext';
 import Join from './pages/Join/Join';
 import Chat from './pages/Chat/Chat';
 
 const App = () => {
   const [name, setName] = useState('');
   const [room, setRoom] = useState('');
   const [socket, setSocket] = useState({});
+  const [showModal, setShowModal] = useState(false);
 
   return (
     <UserContext.Provider value={{
-      name, setName, room, setRoom, socket, setSocket,
+      name, setName, room, setRoom, socket, setSocket, showModal, setShowModal,
```

Comment:
From what I can tell, `Context` looks like a way to pass `props` globally throughout your app. Frankly, I am not sure how React does it but in Angular, we would create different services (plain JS classes) that handle these things for us. For example:

- `UserService` (containing methods to work with the user)
- `RoomService` (containing methods to work with the room)
- `SocketService` (containing methods to work with the current socket)
- `ModalService` (containing methods for opening/closing modals)

Some of these _can_ be global, but consider moving these into separate objects so that this one context does not grow too big.

Edit: It seems like it should be used [sparingly](https://reactjs.org/docs/context.html#before-you-use-context) and there might be other ways to accomplish the same thing in a more "React" way.